src/java/com/rapleaf/jack/queries/GenericQuery.java

```diff
@@ -0,0 +1,149 @@
+package com.rapleaf.jack.queries;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+
+import com.rapleaf.jack.BaseDatabaseConnection;
+import com.rapleaf.jack.ModelField;
+import com.rapleaf.jack.ModelWithId;
+
+public class GenericQuery {
+
+  private final BaseDatabaseConnection dbConnection;
+  private Class<? extends ModelWithId> mainModel;
+  private List<JoinCondition> joinConditions;
+  private List<WhereConstraint> whereConstraints;
```

Comment:
Some of these could be final as well, right? You never allow someone to replace the original list with a new list.